private-templates-service/client/src/store/templateOwner/reducers.ts

```diff
@@ -0,0 +1,64 @@
+import {
+  OwnerState,
+  OwnerType,
+  GET_OWNER_NAME,
+  GET_OWNER_NAME_SUCCESS,
+  GET_OWNER_NAME_FAILURE,
+  GET_OWNER_PROFILE_PICTURE,
+  GET_OWNER_PROFILE_PICTURE_SUCCESS,
+  GET_OWNER_PROFILE_PICTURE_FAILURE,
+  GetOwnerNameAction,
+  GetOwnerProfilePictureAction,
+} from './types';
+
+const initialState: OwnerState = {
+  owner: undefined,
+  isFetching: false,
+}
+
+const initialOwnerState: OwnerType = {
+  displayName: "",
+}
+
+function owner(owner = initialOwnerState, action: GetOwnerNameAction | GetOwnerProfilePictureAction): OwnerType | undefined {
+  switch (action.type) {
+    case GET_OWNER_NAME_SUCCESS:
+      return {
+        ...owner,
+        displayName: action.ownerName
+      }
+    case GET_OWNER_PROFILE_PICTURE_SUCCESS:
+      return {
+        ...owner,
+        imageURL: action.ownerImageURL,
+      }
+    default:
+      return owner;
+  }
+}
+
+export function templateOwnerReducer(state = initialState, action: GetOwnerNameAction | GetOwnerProfilePictureAction): OwnerState {
+  switch (action.type) {
+    case GET_OWNER_NAME:
+    case GET_OWNER_PROFILE_PICTURE:
+      return {
+        ...state,
+        isFetching: true,
+      };
```

Comment:
for case **GET_OWNER_NAME** if there's no code, it will just use the logic of the next case (GET_OWNER_PROFILE_PICTURE)?

Reply:
yes, that's right! the idea is that GET_OWNER_NAME is returning {...state, isFetching: true,} as well

Reply:
alright:) I know it's repetition of code, but wouldn't it be easier to read if you just put case logic there anyway?:)

Reply:
how about adding a comment instead? :)

Reply:
sure:)